.github/workflows/build-pulse.yml

```
@@ -0,0 +1,60 @@
+name: BuildPulse
+
+on:
+  workflow_run:
+    workflows: ["Test"]
+    types:
+      - completed
+  workflow_dispatch:
+    inputs:
+      artifacts_url:
+        description: The URL of the artifacts for the Test workflow run
+        required: true
+      head_commit_id:
+        description: The commit ID of the head commit of the Test workflow run
+        required: true
+
+defaults:
+  run:
+    shell: bash
+
+permissions:
+  actions: read
+
+jobs:
+  buildpulse:
+    name: Upload JUnit XML report to BuildPulse
+    runs-on: ubuntu-latest
+    if: github.event.workflow_run.conclusion == 'success' || github.event.workflow_run.conclusion == 'failure'
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.inputs.head_commit_id || github.event.workflow_run.head_commit.id }}
+      - id: reports
+        run: mktemp -d | xargs -0 -I{} echo "path={}" | tee -a $GITHUB_OUTPUT
+      - name: Download all JUnit XML reports from the Test workflow run
```

```suggestion
      - name: Download the artifacts so can access the JUnit XML reports from the Test workflow run
```

Nit: but trying to be more accurate to what is actually happening